### ISSUE TITLE
Shorter/Improved remapped names

### ISSUE THREAD

**Issue by solonovamax** [FabricMC/fabric-loom#1164]
Currently, when mods are remapped by loom, they result in names that look like the following:
```
loom_mappings_1_19_4_layered_hash_40359_v2.net.fabricmc.fabric-api:fabric-data-generation-api-v1:11.5.2+26f09d40f4
```
This name is quite long and unwieldy, and is annoying to look at in your IDE when attempting to find the sources for a particular library.

I would like to propose that instead when remapping, the `loom_mappings_1_19_4_layered_hash_40359_v2` bit can be replaced with, for example, a 6 character MD5 hash of the minecraft version & the mappings used (ofc properly accounting for layered mappings). eg. minecraft 1.21.1 using yarn mappings version `1.21.1+build.3` might result in the following remapped mod name:
```
remapped_cd6b1a.net.fabricmc.fabric-api:fabric-data-generation-api-v1:11.5.2+26f09d40f4
```
`cd6b1a` is derived from the first 6 characters of the MD5 hash of `minecraft-1.21.1:net.fabricmc-yarn-1.21.1+build.3`. `minecraft-[version]` is used to indicate the minecraft version, and `[mappings group]-[mappings module]-[mappings version]` is used for the mappings used. They are concatenated with a `:` to separate them.

another alternative to prefixing the hash in front of the group id would be to prefix it in front of the version:
```
net.fabricmc.fabric-api:fabric-data-generation-api-v1:remapped-cd6b1a.11.5.2+26f09d40f4
```
this would keep some of the more important information at the front, such as the group id and module name.

**Comment by modmuss50** [comment]
Yes, I I do think we can improve this. We do need to make sure that the group is diffrent, it could maybe even be something as small as.

`abc123.net.fabricmc.fabric-api:fabric-data-generation-api-v1:11.5.2+26f09d40f4` 🤔

**Comment by solonovamax** [comment]
> @modmuss50
> Yes, I I do think we can improve this. We do need to make sure that the group is diffrent, it could maybe even be something as small as.
> 
> `abc123.net.fabricmc.fabric-api:fabric-data-generation-api-v1:11.5.2+26f09d40f4` 🤔

why specifically the group id?

**Comment by solonovamax** [comment]
also tbh I was originally gonna put smth like that but I wasn't sure if it was clear enough where the hash was coming from

**Comment by solonovamax** [comment]
from what I can tell (by looking at the source, I have not yet tested the loom 1.11 snapshots), the names are now going to look something like this:
```
remapped.net.fabricmc.fabric-api:fabric-data-generation-api-v1-[mapping name].[mc version].[version][classifier]:11.5.2+26f09d40f4
```
eg.
```
remapped.net.fabricmc.fabric-api:fabric-data-generation-api-v1-net.fabricmc.yarn.1_19_4.1.19.4+build.1-v2:11.5.2+26f09d40f4
```
correct?

this is quite a bit better than before because you can at least see the group id & name without having a huge sidebar.


however, I would *prefer* (though I _wouldn't_ call it an "problem", and thus won't bother opening a new issue/requesting this one to be re-opened, rather this is just a preference) something like:
```
remapped.net.fabricmc.fabric-api:fabric-data-generation-api-v1-[hash]:11.5.2+26f09d40f4
```
where `[hash]` is a 10-character hash (could be sha256 or anything else, I don't think it really matters) of the mappings group, mappings name, mappings version, and mappings classifier (so like `sha256("${mappings.group}:${mappings.name}:${mappings.version}:${mappings.classifier}")` or smth. if using multiple mappings, just concat all of them and then hash it.). eg.
```
remapped.net.fabricmc.fabric-api:fabric-data-generation-api-v1-abcde12345:11.5.2+26f09d40f4
```

this way, the full name is rather short.
is there any reason that this specific format was chosen over just using a hash? if it's for clashes on the filesystem, then afaik you can still have the folder it's placed in use the mapping group, name, and version, but then have the actual group & name of the dependency be something other than the folder(s) it's placed in, but I *could* be wrong.

**Comment by modmuss50** [comment]
> from what I can tell (by looking at the source, I have not yet tested the loom 1.11 snapshots), the names are now going to look something like this:

No, I also added the hash. The main change in my PR was to use a 32bit hash in place of the mapping name, so your proposed example is what we now have:
```
remapped.net.fabricmc.fabric-api:fabric-data-generation-api-v1-12345678:11.5.2+26f09d40f4
```

For the record, we must change the start of the group the `remapped.` bit to not cause issues when exclusiveContent is set.

**Comment by solonovamax** [comment]
> > from what I can tell (by looking at the source, I have not yet tested the loom 1.11 snapshots), the names are now going to look something like this:
> 
> No, I also added the hash. The main change in my PR was to use a 32bit hash in place of the mapping name, so your proposed example is what we now have:
> 
> ```
> remapped.net.fabricmc.fabric-api:fabric-data-generation-api-v1-12345678:11.5.2+26f09d40f4
> ```
> 
> For the record, we must change the start of the group the `remapped.` bit to not cause issues when exclusiveContent is set.

ah, my bad. must have missed smth when reading the source code.

yeah, I get that, and have no issue with that
